task.go

```diff
@@ -294,14 +302,32 @@ func (t *task) Exec(ctx context.Context, id string, spec *specs.Process, ioCreat
 	}, nil
 }
 
-func (t *task) Pids(ctx context.Context) ([]uint32, error) {
+func (t *task) Pids(ctx context.Context) ([]ProcessInfo, error) {
 	response, err := t.client.TaskService().ListPids(ctx, &tasks.ListPidsRequest{
 		ContainerID: t.id,
 	})
 	if err != nil {
 		return nil, errdefs.FromGRPC(err)
 	}
-	return response.Pids, nil
+	var processList []ProcessInfo
+	for _, p := range response.Processes {
+		processInfo := ProcessInfo{
+			Pid: p.Pid,
+		}
+		if p.Info != nil {
+			v, err := typeurl.UnmarshalAny(p.Info)
+			if err != nil {
+				return nil, errdefs.FromGRPC(err)
+			}
+			t, ok := v.(interface{})
```

Comment:
is this actually needed?

Reply:
I wasn't sure - I left the type assertion in case that the service returned a non-interface type. Thoughts?

Reply:
Since, `UnmarshalAny` returns an `interface{}` I don't think its possible to not have an interface type until it returns a `nil` and that won't happen without an error.

Reply:
Gotcha, makes sense, I'll update!